float64 target_distance
float64 target_angle
int64 process_flag